presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -157,6 +159,35 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         return new HivePartitionResult(partitionColumns, partitions.build(), remainingTupleDomain, enforcedTupleDomain, hiveBucketHandle);
     }
 
+    public Optional<DiscretePredicates> getTableDiscretePredicates(SemiTransactionalHiveMetastore metastore, ConnectorTableHandle tableHandle)
+    {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
+        SchemaTableName tableName = hiveTableHandle.getSchemaTableName();
+        Table table = getTable(metastore, tableName);
+
+        List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(connectorId, table);
+        List<Type> partitionTypes = partitionColumns.stream()
+                .map(column -> typeManager.getType(column.getTypeSignature()))
+                .collect(toList());
+
+        List<Map<ColumnHandle, NullableValue>> partitionKeys = new ArrayList<>();
+        List<String> partitionNames = getFilteredPartitionNames(metastore, tableName, partitionColumns, TupleDomain.all());
```

Comment:
For query like `SHOW PARTITIONS FROM .. WHERE ..`, we should construct a filtering `TupleDomain`, so that we pull only relevant partitions from the metastore.